src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +158,1547 @@ ActorInit En_Jso_InitVars = {
+        (this->action != EN_JSO_ACTION_UNK_15)) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->bodyCollider.base);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->bodyCollider.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AECA8.s")
+        if ((this->action == EN_JSO_ACTION_SLASH) && !this->swordHitSomething &&
+            (this->swordState == EN_JSO_SWORD_STATE_BOTH_DRAWN)) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->rightSwordCollider.base);
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->leftSwordCollider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AED00.s")
+s32 EnJso_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnJso* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AED54.s")
+    if (limbIndex == GARO_LIMB_RIGHT_ARM) {
+        rot->x += this->rightArmRot.x;
+        rot->y += this->rightArmRot.y;
+        rot->z += this->rightArmRot.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AEDAC.s")
+    if (limbIndex == GARO_LIMB_ROBE_RIGHT) {
+        rot->x += this->robeRightRot.x;
+        rot->y += this->robeRightRot.y;
+        rot->z += this->robeRightRot.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AEE44.s")
+    if ((limbIndex == GARO_LIMB_LEFT_SWORD) || (limbIndex == GARO_LIMB_RIGHT_SWORD)) {
+        if (this->swordState == EN_JSO_SWORD_STATE_NONE_DRAWN) {
+            *dList = NULL;
+        } else if (limbIndex == GARO_LIMB_LEFT_SWORD) {
+            if (this->swordState == EN_JSO_SWORD_STATE_RIGHT_DRAWN) {
+                *dList = NULL;
+            }
+        } else if ((limbIndex == GARO_LIMB_RIGHT_SWORD) && (this->swordState == EN_JSO_SWORD_STATE_LEFT_DRAWN)) {
+            *dList = NULL;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AEEC0.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AF064.s")
+void EnJso_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f sSwordTipOffset = { 1600.0f, 0.0f, 0.0f };
+    static Vec3f sSwordBaseOffset = { 0.0f, 0.0f, 0.0f };
+    static Vec3f sSwordTipQuadOffset = { 1700.0f, 0.0f, 0.0f };
+    static Vec3f sSwordBaseQuadOffset = { 0.0f, 0.0f, 0.0f };
+    EnJso* this = THIS;
+    Vec3f swordTipPos;
+    Vec3f swordBasePos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AF110.s")
+    Matrix_Push();
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Jso/func_809AF28C.s")
+    if (limbIndex == GARO_LIMB_LEFT_SWORD) {
+        if (this->swordState == EN_JSO_SWORD_STATE_KNOCKED_OUT_OF_HANDS) {
+            Actor_SpawnBodyParts(&this->actor, play, 15, dList);
```

Comment:
```suggestion
            Actor_SpawnBodyParts(&this->actor, play, ENPART_TYPE_15, dList);
```
Might need to include `z_en_part.h` here

Reply:
Huh, wondered about this one, because *none* of the calls to `Actor_SpawnBodyParts` are using this enum, so I figured it must not exist. Let me update them to all use the enum.

Reply:
Yeah, I noticed that too. I was going to do that in my next anim cleanup PR but it'd be great if you wanted to do that right now. A step further would be to also make a param setter macro like clear-tag, but I'll leave it up to you for how far you want to take

Reply:
I just made it like EnClearTag, however I did not bother to name the part types so they're still `ENPART_TYPE_15` for example. I figure they can be named in a follow-up PR; changing what other actors pass into `Actor_SpawnBodyParts` already expands the scope of this PR a bit, so let's not make it any bigger.